File: FitToFit/UI/fittofit_admin/lib/models/akcijeTreninzi.dart
import 'package:fittofit_admin/models/treninzi.dart';
import 'package:json_annotation/json_annotation.dart';

part 'akcijeTreninzi.g.dart';

@JsonSerializable()
class AkcijeTreninzi {
  int akcijaTreningId;
  int akcijaId;
  int treningId;
  Treninzi? trening;

  AkcijeTreninzi(
      {required this.akcijaTreningId,
      required this.akcijaId,
      required this.treningId,
      this.trening});

  factory AkcijeTreninzi.fromJson(Map<String, dynamic> json) =>
      _$AkcijeTreninziFromJson(json);

  Map<String, dynamic> toJson() => _$AkcijeTreninziToJson(this);
}

File: FitToFit/FitToFit.Services/Database/AkcijeTreninzi.cs
using System;
using System.Collections.Generic;

namespace FitToFit.Database;

public partial class AkcijeTreninzi
{
    public int AkcijaTreningId { get; set; }

    public int AkcijaId { get; set; }

    public int TreningId { get; set; }

    public virtual Akcije Akcija { get; set; } = null!;

    public virtual Treninzi Trening { get; set; } = null!;
}
